### ISSUE TITLE
Stability of positivity constraint

### ISSUE THREAD

**Issue by patel-zeel** [wesselb/varz#2]
Hi @wesselb,

If I am getting it right, Varz handles positivity constraints with `exp(x)` transformation which could be explosive for large values. How about adding more stable transformations for positivity such as `log(1 + exp(x))` (softplus)?

For GP models, `log(1 + exp(x))` seems popular and more stable as per the following references.
1. [GPy](https://github.com/SheffieldML/GPy/blob/3e19a85575687e37fd6f61174115d7c94d2c96e6/GPy/kern/src/stationary.py#L78), [paramz](https://github.com/sods/paramz/blob/897fb8b92f51c17e1dd92c60f69df718a3899d1b/paramz/transformations.py#L105)
2.  [GPFlow positive bijector](https://github.com/GPflow/GPflow/blob/d088c2c4583df606a6f0ba79eea78d27c6210f7c/gpflow/config/__config__.py#L173)
3. [`gpytorch.constraints.Positive`](https://docs.gpytorch.ai/en/v1.1.1/_modules/gpytorch/constraints/constraints.html#Positive)

**Comment by wesselb** [comment]
Hey @patel-zeel,

Good observation! In my experience, if you can use `exp` without it NaNing out, it gives faster convergence than `softplus`. However, you're right that `exp` doesn't work for all use cases. I see a few options:

1. Change `exp` to `softplus`.
2. Add a `method` keyword which can be set to `"exp"` or `"softplus"` (`vs.positive(method="exp")`). Make the default `"exp"`.
3. Do 2, but make the default `"softplus"`.

Since convergence tends to be quicker with `exp` and `exp` usually (though definitely not always) seems to be fine, I'm tending towards 2. What are your thoughts?

**Comment by patel-zeel** [comment]
I agree with Point 2. It is nice to know that `exp` converges faster than `softplus`. I will surely try it out in my future experiments. 

Can I try adding Point 2 with a PR?

**Comment by wesselb** [comment]
> Can I try adding Point 2 with a PR?

Definitely. :) That would be amazing!
